src/Voron/Data/DirectAddScope.cs

```diff
@@ -5,35 +5,45 @@ namespace Voron.Data
 {
     public unsafe class DirectAddScope : IDisposable
     {
-        private readonly ITree _tree;
+        private readonly string _treeName;
```

Comment:
Just pass the parent tree instance directly (as `object`) and call `ToString()` only if we need the error.